packages/myst-cli/src/process/search.ts

```diff
@@ -0,0 +1,152 @@
+import type { GenericNode } from 'myst-common';
+import type { DocumentHierarchy } from 'myst-spec-ext';
+import type { Heading } from 'myst-spec';
+import { toText } from 'myst-common';
+export const SKIP = () => {};
+
+type Visit = (content: GenericNode, path: string) => typeof SKIP | void;
+type Depart = (content: GenericNode, path: string) => void;
+
+function visitNode(content: GenericNode, visit: Visit, depart: Depart | undefined, path: string) {
```

Comment:
This stuff is all hand-rolled visitor logic. Maybe we can refactor it now to use the unist utilities. At one point it did more work.

Reply:
Yeah - this looks identical to the visit code. I think it is worth factoring it out and using the existing tools.